### PR TITLE
bump rocprofiler-dev to 6.0.2 and add aqlprofile

### PR DESCRIPTION
**Pull request by afzpatel**
aqlprofile is required for rocprofiler-dev. The aqlprofile are extracted from the yum, zyp or apt packages and installed to the prefix path.

### CODE REVIEW

File: var/spack/repos/builtin/packages/aqlprofile/package.py
Comment (afzpatel) [inline var/spack/repos/builtin/packages/aqlprofile/package.py:1]: using localhost as a placeholder for aql url until a public location is determined

**Review by srekolam — APPROVED** [review]

**Review by renjithravindrankannath — APPROVED** [review]
LGTM

**Comment by afzpatel** [comment]
Hi @alalazo @tldahlgren this pr is ready to be reviewed.

**Review by tldahlgren — APPROVED** [review]
Confirmed the new version sha256s for `rocprofiler-dev` and a couple of `aqlprofile`.  

Would it be appropriate to add a homepage property to `aqlprovile` (in another PR)?

**Comment by afzpatel** [comment]
> Confirmed the new version sha256s for `rocprofiler-dev` and a couple of `aqlprofile`.
> 
> Would it be appropriate to add a homepage property to `aqlprovile` (in another PR)?

Hmmm.... I can't really think of anything that would be appropriate to add as a homepage for aqlprofile. It's not open source so there is no github page for it, nor is there any public documentation for it. But if I find something appropriate, I'll add it in a future pr.

**Comment by mwkrentel** [comment]
@afzpatel This patch breaks the build for `hpctoolkit @develop +rocm` because
there is no `hpcrun/Makefile.in` file.  Develop uses meson, not autotools.

For meson, you'd have to add cpp flags in one of the meson.build files, I guess.

Can you elaborate on why you need to add `-D__HIP_PLATFORM_AMD__=1`
to `CPP_FLAGS` for hpctoolkit?

Can you show me some way to reproduce something that fails without this?

AFAICT, the source files already define `__HIP_PLATFORM_AMD__`.
For example, `src/hpcrun/gpu/api/amd/hip-api.c` has:
```
#define _GNU_SOURCE
#define __HIP_PLATFORM_AMD__
#define __HIP_PLATFORM_HCC__

#include <dlfcn.h>
#include <stdio.h>
#include <string.h>    // memset

#include <hip/hip_runtime.h>
```

And indeed, I tried commenting out the patch, and for me, both
hpctoolkit 2024.01.1 and develop built cleanly with +rocm and hip 6.0.0.

Btw, I wonder why spackbot didn't notify me about this PR.

**Pull request merged**